### PR TITLE
Object selection

### PR DESCRIPTION
**Pull request by radenui**
Hi, 
With this PR, it adds ability to select the objects you can instersect with mouse-cursor.
`mouse-cursor="objects: .hoverable; recursive: false"` will only intersect objects with class `hoverable`.
- Selection of objects is done in the scene
- Default objects: '*' (reproduces the default behaviour)
- Default recursive: false

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -27,14 +30,17 @@ AFRAME.registerComponent('mouse-cursor', {
...
   },
 
   /**
    * Called when component is attached and when component data changes.
    * Generally modifies the entity based on the data.
    * @protected
    */
-  update (oldData) { },
+  update (oldData) {
+    this.__refreshObjects('update');
```
Comment (ip) [inline index.js:42]: Why is the argument needed?

File: index.js [hunk 2]
```diff
@@ -420,4 +431,43 @@ AFRAME.registerComponent('mouse-cursor', {
...
+        this.__addElement(obj, recursive);
+      };
+    }
+  },
+
+  /**
+   * @private
+   */
+  __isElementInObjects(el) {
+    for (var i = 0; i < this.objects.length; i++) {
```
Comment (ip) [inline index.js:467]: Can be replaced by `return el in this.objects`

**Comment by ip** [comment]
Refreshing the list of filtered objects on every `child-attached`/`child-detached` event is unnecessary and expensive, as they're bubbled and there may be lots of them per render frame. It's better to use dirty flag pattern (lazy invocation).

Here is an updated version of this pull request: https://github.com/ip/aframe-mouse-cursor-component/tree/object_selection_2

**Comment by mayognaise** [comment]
Hi @ip, this feature is now available in A-Frame v0.6.1 by setting <a-scene cursor="rayOrigin: mouse">. 🎉

**Pull request closed without merging**